torchgeo/datasets/solar_plants_brazil.py

```diff
@@ -0,0 +1,230 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""SolarPlantsBrazil dataset."""
+
+import glob
+import os
+from collections.abc import Callable
+from typing import Literal
+
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio
+import torch
+from matplotlib.figure import Figure
+from torch import Tensor
+
+from .errors import DatasetNotFoundError
+from .geo import NonGeoDataset
+from .utils import Path, download_and_extract_archive
+
+
+class SolarPlantsBrazil(NonGeoDataset):
+    """Solar Plants Brazil dataset (semantic segmentation for photovoltaic detection).
+
+    The `Solar Plants Brazil <https://huggingface.co/datasets/FederCO23/solar-plants-brazil>`__
+    dataset provides satellite imagery and pixel-level annotations for detecting photovoltaic
+    solar power stations.
+
+    Dataset features:
+
+    * 272 RGB+NIR GeoTIFF images (256x256 pixels)
+    * Binary masks indicating presence of solar panels (1 = panel, 0 = background)
+    * Organized into `train`, `val`, and `test` splits
+    * Float32 GeoTIFF files for both input and mask images
+    * Spatial metadata included (CRS, bounding box), but not used directly for training
+
+    Folder structure:
+
+    .. code-block:: text
+
+        root/train/input/img(123).tif
+        root/train/labels/target(123).tif
+
+    Access:
+
+    * Dataset is hosted on Hugging Face: https://huggingface.co/datasets/FederCO23/solar-plants-brazil
+    * Code and preprocessing steps available at: https://github.com/FederCO23/UCSD_MLBootcamp_Capstone
+
+    .. versionadded:: 0.8
+
+    """
+
+    url = 'https://huggingface.co/datasets/FederCO23/solar-plants-brazil/resolve/main/solarplantsbrazil.zip'
+    bands = ('Red', 'Green', 'Blue', 'NIR')
+    md5 = 'DFA0D3EFDEF4143A33B0E7BA834EAAFA'
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        split: Literal['train', 'val', 'test'] = 'train',
+        transforms: Callable[[dict[str, Tensor]], dict[str, Tensor]] | None = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a SolarPlantsBrazil dataset split.
+
+        Args:
+            root: Root directory where dataset is stored.
+            split: Dataset split to use, one of "train", "val", or "test".
+            transforms: Optional transforms to apply.
+            download: If True, download the dataset if it doesn't exist.
+            checksum: If True, verify MD5 checksum of the downloaded file.
+
+        Raises:
+            DatasetNotFoundError: If the dataset is not found and
+                ``download=False``.
+
+        Returns:
+            None
+
+        """
+        if split not in ['train', 'val', 'test']:
+            raise ValueError(
+                f"Invalid split '{split}', expected one of: 'train', 'val', or 'test'"
+            )
```

Comment:
Could honestly remove this and the test since we statically check that a valid Literal is used, but up to you. This is certainly nicer for users as it gives them a more clear explanation and is checked at runtime.